### PR TITLE
[pt] Added (style) rule ID:NOMECOMUM_PRONOME_VERBINF_REM_PRON

### PR DESCRIPTION
**Pull request by marcoagpinto**
@susanaboatto @p-goulart 

Heya,

My first pull request.

Feel free to make changes if they are required.

Thanks!

### CODE REVIEW

**Comment by maphjo** [comment]
Thank you for trying the PR system! 
This is really going to help us understand if the `toneTag` definitions are working.
@susanaboatto or @p-goulart will have a look at the rule after the weekend. ;-)

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml [hunk 1]
```diff
@@ -11774,7 +11774,47 @@ USA
...
     <category id='REMOVE_PRONOUN' name='Remoção de pronomes' type='style' tone_tags="povrem">
+
+
+        <!-- POR ELE SABER por saber -->
+        <rule id='NOMECOMUM_PRONOME_VERBINF_REM_PRON' name="Simplificar: Nome Comum + Pronome + Verbo Infinitivo → Omitir Pronome" type='style' tone_tags="povrem">
+            <!-- Created by Marco A.G.Pinto with Ricardo Joseh Lima suggestions, Portuguese rule 2023-07-08 (8-JUL-2023+) -->
+            <!--
+            O rei envia o mensageiro com rapidez por ele saber que o tempo é curto. → O rei envia o mensageiro com rapidez por saber que o tempo é curto.
+            Os reis enviam o mensageiro com rapidez por eles saberem que o tempo é curto. → Os reis enviam o mensageiro com rapidez por saberem que o tempo é curto.
+            -->
```
Comment (p-goulart) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml:11788]: Personally, I don't think we need these types of comments, as they don't add much new information. Your examples are already tagged and tested as `example`s, and we can check the authorship of specific rules with a simple `git blame`.

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml [hunk 2]
```diff
@@ -11774,7 +11774,47 @@ USA
             </rule>
         </rulegroup>
     </category>
+
+
     <category id='REMOVE_PRONOUN' name='Remoção de pronomes' type='style' tone_tags="povrem">
+
+
+        <!-- POR ELE SABER por saber -->
+        <rule id='NOMECOMUM_PRONOME_VERBINF_REM_PRON' name="Simplificar: Nome Comum + Pronome + Verbo Infinitivo → Omitir Pronome" type='style' tone_tags="povrem">
```
Comment (p-goulart) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml:11783]: Not sure if you've seen it, but I've already been replacing all instances of 'nome' with 'substantivo'. The former isn't really used to mean 'noun' in pt-BR, so using 'substantivo' should be more inclusive. In fact, I'm not entirely sure myself why the rule ID makes reference to common nouns, since the regexp used to define the noun token specifically allows for `NP`.

File: languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml [hunk 3]
```diff
@@ -11774,7 +11774,47 @@ USA
...
+                <token postag='SPS00|(SPS00:)?[DP][ADIPRT].+|VMIP.+|_PUNCT' postag_regexp='yes'/>
+                <token postag='NC.+|AQ.+|NP.+' postag_regexp='yes'/>
+                <marker>
+                    <token>por</token>
+                    <token regexp='yes'>el[ae]s?|est[ae]s?</token>
+                    <token postag='VMN03.+' postag_regexp='yes'/>
+                </marker>
+                <token><exception postag_regexp='yes' postag='VMG0000|SENT_END'/></token>
+            </pattern>
+            <message>Poderá omitir o pronome nesta perífrase.</message>
```
Comment (p-goulart) [inline languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml:11811]:
```suggestion
            <message>Para um texto mais sucinto, prefira omitir o pronome.</message>
```

**Review by p-goulart — CHANGES_REQUESTED** [review]

**Comment by marcoagpinto** [comment]
@p-goulart 

Sorry for only now replying, but the Internet failed in the morning, and then I had a date and only now I arrived back home.

Could I ask you, please, if you could change the rule directly like Susana used to do with my rules?

In the next ones I will try to do better.

In Portugal, we call them "substantivo" too, but then Priberam and other dictionaries started calling them "Nome Comuns".

Personally, I like "substantivo" more, but I thought that it was some new agreement/convention regarding Portuguese talking countries.

Thanks!

**Comment by p-goulart** [comment]
Hey, please don't worry about your reply time – no one should expect anyone to react **immediately**. And you shouldn't consider my comments as having to do 'better' – we're just trying to keep things more aligned and consistent for _everyone_.

And, I mean, I suppose we _could_ edit your rules, but that's not really the _point_ of code review. We're not in a huge hurry, so I think we can take the time to discuss and polish everything up before we merge it, right?

**Comment by marcoagpinto** [comment]
@p-goulart 

Ahhhh... thank you, my brother,

I will try to fix it tonight.

❤️ ❤️ ❤️ ❤️ ❤️

**Comment by marcoagpinto** [comment]
@p-goulart 

Heya,

I need your help.

In TortoiseGit I selected “amend” + “recommit” and it seems it committed, but I don't see the patch anywhere.

Then I tried again and changed the branch, and still, it doesn't appear anywhere.

And now it says there is nothing to commit.

What shall I do?

Thanks!

**Comment by marcoagpinto** [comment]
Here is what I amended+recommitted:

```
        <rule id='SUBSTANTIVO_PRONOME_VERBINF_REM_PRON' name="Simplificar: Substantivo + Pronome + Verbo Infinitivo → Omitir Pronome" type='style' tone_tags="clarity">
            <!-- Created by Marco A.G.Pinto with Ricardo Joseh Lima suggestions, Portuguese rule 2023-07-08 (8-JUL-2023+) -->
            <antipattern>
                <token postag='SENT_START|CC|CS' postag_regexp='yes'/>
                <token min='0' max='1' postag='_PUNCT|_QUOT' postag_regexp='yes'/>
                <token postag='V.+' postag_regexp='yes'/>
                <token postag='SPS00|(SPS00:)?[DP][ADIPRT].+|VMIP.+|_PUNCT' postag_regexp='yes'/>
                <token postag='NC.+|AQ.+|NP.+' postag_regexp='yes'/>
                <token>por</token>
                <token regexp='yes'>el[ae]s?|est[ae]s?</token>
                <token postag='VMN03.+' postag_regexp='yes'/>
                <example>Drymon disse que "Tivemos muita sorte por ela ter tido o contato, caso contrário não poderíamos usá-lo.</example>
                <example>...linson, registrou o quanto Ellen lutou para evitar o adiamento da reforma, e expressou sua tristeza por ela ter morrido antes da data marcada.</example>
            </antipattern>
            <pattern>
                <token postag='SPS00|(SPS00:)?[DP][ADIPRT].+|VMIP.+|_PUNCT' postag_regexp='yes'/>
                <token postag='NC.+|AQ.+|NP.+' postag_regexp='yes'/>
                <marker>
                    <token>por</token>
                    <token regexp='yes'>el[ae]s?|est[ae]s?</token>
                    <token postag='VMN03.+' postag_regexp='yes'/>
                </marker>
                <token><exception postag_regexp='yes' postag='VMG0000|SENT_END'/></token>
            </pattern>
            <message>Para um texto mais sucinto, prefira omitir o pronome.</message>
            <suggestion>\3 \5</suggestion>
            <example correction="por saber">O rei envia o mensageiro com rapidez <marker>por ele saber</marker> que o tempo é curto.</example>
            <example correction="por saberem">Os reis enviam o mensageiro com rapidez <marker>por eles saberem</marker> que o tempo é curto.</example>
        </rule>
```

I kept this line:
`<!-- Created by Marco A.G.Pinto with Ricardo Joseh Lima suggestions, Portuguese rule 2023-07-08 (8-JUL-2023+) -->`

Because it was Ricardo (PhD in Linguistics) who said the results were good. It is he who comments on rules I suggest and create.

Also, it has the date the rule was created plus the `(8-JUL-2023+)` which represents the day of the new knowledge I started to use.

If you notice, all my rules have the date of the new knowledge I got, which allows me to update older rules I created once my knowledge grows.

So, it is vital information.

Thanks!

**Comment by p-goulart** [comment]
> In TortoiseGit I selected “amend” + “recommit” and it seems it committed, but I don't see the patch anywhere.
> Then I tried again and changed the branch, and still, it doesn't appear anywhere.
> And now it says there is nothing to commit.
> What shall I do?

I don't really use TortoiseSVN, so I can't tell you. When using git, one would usually not do an `amend` here, as an amend operation on a commit that's already on the remote would be rejected (you'd need to force-push). Can you not simply add another commit to the same branch (viz. `lt_marcoagpinto_20230708_0727`)? Sounds simpler.

Contributing over Git would be preferable, as that's the standard for this project, and I'm experienced with Git and would be able to walk you through the necessary steps. But if it *must* be TortoiseSVN I'm afraid I can't be of much help :/

---

As for keeping the comment line, I'm afraid I must disagree again.

If doing this *in this particular format* is vital to **you** then perhaps you should keep a **local** record of what rules you added when once they're merged. Since we're working on a Git project **none** of that information is **ever** lost. We can simply do `git blame` on the line where a rule is defined and see who created the rule and when.

Now, since you use TortoiseSVN I understand it may not work quite the same way, but I'd be very surprised if there wasn't a way to retrieve that information. Some modern text editors have very nifty Git plugins that enable one to visualise effortlessly the authorship of individual lines of code as well. The built-in Git functionality of IntelliJ, for example, is quite extensive, cf.:

Screenshot 2023-07-11 at 9 20 56 AM

I understand that's a bit of a departure from how you've done things so far, but as we expand the team we need to clean up some of our git practices. These files are already massive and somewhat messy, and these types of comments add a lot of noise. If the comment adds information that isn't already available in the Git metadata, however, then of course you are encouraged to leave a comment.

**Comment by marcoagpinto** [comment]
Ahhhh... I used TortoiseGit, not TortoiseSVN.

I am still learning the basics.

Let me try again.

**Comment by p-goulart** [comment]
Oops, yes, my mistake, TortoiseGit. I can't even install it here and try to repro whatever problems you're going through as I'm on macOS 🍏

For Windows, I **have** had to use [Git BASH](https://gitforwindows.org) in the past on occasion, but that would require some familiarity with bash.

You can also use the Git interface that comes with IntelliJ – that's what many people on the team use. Out of curiosity, what text editor do you use for the grammar XMLs?

**Comment by marcoagpinto** [comment]
Pedro,

I use what all developers use: Notepad ++

Well, TortoiseGit gives an error when I attempt to commit to the old branch.

I will refresh the repository in my harddisk and will create a new branch.

Sorrry :(

**Review by p-goulart — APPROVED** [review]

**Comment by p-goulart** [comment]
Your changes are approved, @marcoagpinto , don't worry about it. Do you want to merge them yourself or should I?

**Comment by marcoagpinto** [comment]
Could you merge them, please?

Always remember that both you and Susana are always free to change the rules the way they work better.

Some are like raw diamonds which need to be polished.

Thank you!

❤️ ❤️ ❤️
